editor/plugins/animation_blend_tree_editor_plugin.cpp

```diff
@@ -1114,10 +1131,20 @@ AnimationNodeBlendTreeEditor::AnimationNodeBlendTreeEditor() {
 	add_options.push_back(AddOption("TimeSeek", "AnimationNodeTimeSeek", 1));
 	add_options.push_back(AddOption("TimeScale", "AnimationNodeTimeScale", 1));
 	add_options.push_back(AddOption("Transition", "AnimationNodeTransition"));
-	add_options.push_back(AddOption("BlendTree", "AnimationNodeBlendTree"));
-	add_options.push_back(AddOption("BlendSpace1D", "AnimationNodeBlendSpace1D"));
-	add_options.push_back(AddOption("BlendSpace2D", "AnimationNodeBlendSpace2D"));
-	add_options.push_back(AddOption("StateMachine", "AnimationNodeStateMachine"));
+	
+	//add all inheriters of AnimationRootNodes types to options
+	List<StringName> classes;
+	classes.sort_custom<StringName::AlphCompare>();
+	ClassDB::get_inheriters_from_class("AnimationRootNode", &classes);
+
+	for (List<StringName>::Element *E = classes.front(); E; E = E->next()) {
+		String name = String(E->get()).replace_first("AnimationNode", "");
+		if (name == "Animation" || name == "StartState" || name == "EndState") {
+			continue; // nope
+		}
+		add_options.push_back(AddOption(vformat(TTR("%s"), name), E->get()));
```

Comment:
```suggestion
		add_options.push_back(AddOption(name, E->get()));
```
This doesn't work, or rather it is pointless, the translation just translates "%s", not the name, also we shouldn't translate here as we don't do that for the rest